### PR TITLE
hal_nxp: Zephyr port on iMx8MNano (M7) using Digi CC8MNano DVK

### PR DESCRIPTION
**Pull request by waseemh40**
Zephyr port on iMx8MNano processor using Digi CC8MNano DVK. NXP Hal directory.

Signed-off-by: Waseem Hassan <waseemh40@gmail.com>

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
1. please drop the middleware/ code for now.  We have openamp support in the tree that provides this.
2. split this into at least 2 commits, one that adds the imported files, the other that makes any cmake/readme/etc changes.

**Comment by waseemh40** [comment]
Hello,

I have modified the repo as per requested changes. Hope it is OK now.

-Waseem

**Comment by waseemh40** [comment]
Hello,

@galak Here also, what happens next? I have added the requested modfications.

-Waseem

File: mcux/CMakeLists.txt [hunk 1]
```diff
@@ -61,6 +61,16 @@ add_subdirectory_ifdef(
   middleware/wireless/framework_5.3.3
   )
 
+if (${MCUX_DEVICE} MATCHES "MIMX8MN6")
+	zephyr_library_sources_ifdef(CONFIG_IPM_IMX_MCUX	drivers/imx8/fsl_mu.c)
+	zephyr_library_sources(drivers/imx8/fsl_gpc.c)
+	zephyr_library_sources(devices/${MCUX_DEVICE}/system_MIMX8MN6_cm7.c)
+	add_subdirectory_ifdef(
```
Comment (galak) [inline mcux/CMakeLists.txt:68]: remove the rpmsg_lite reference here

**Review by galak — CHANGES_REQUESTED** [review]
Can you remove mcux/middleware dir as I don't think its needed.

**Comment by galak** [comment]
> Hello,
> 
> @galak Here also, what happens next? I have added the requested modfications.
> 
> -Waseem

I was off on vacation less week.

**Comment by waseemh40** [comment]
Hi,

I have fixed the CMake file. Also, I have removed the middleware directoy.

-Waseem

File: mcux/CMakeLists.txt [hunk 2]
```diff
@@ -61,6 +61,12 @@ add_subdirectory_ifdef(
   middleware/wireless/framework_5.3.3
   )
 
+if (${MCUX_DEVICE} MATCHES "MIMX8MN6")
+	zephyr_library_sources_ifdef(CONFIG_IPM_IMX_MCUX	drivers/imx8/fsl_mu.c)
+	zephyr_library_sources(drivers/imx8/fsl_gpc.c)
```
Comment (galak) [inline mcux/CMakeLists.txt:66]: fsl_gpc.c and fsl_mu.c should be handled in `mcux/drivers/imx8/CMakeLists.txt` not here.

**Review by galak — COMMENTED** [review]
Sorry, I didn't realize mcux/middleware/wireless and mcux/middleware/usb already existed in the tree.  As such, please don't remove them.

**Comment by waseemh40** [comment]
Restored usb and wireless. Also removed fsl_gpc and fsl_mu from the CMake file since they are not required for UART and GPIO operations.

File: mcux/CMakeLists.txt [hunk 3]
```diff
@@ -61,6 +61,10 @@ add_subdirectory_ifdef(
   middleware/wireless/framework_5.3.3
   )
 
+if (${MCUX_DEVICE} MATCHES "MIMX8MN6")
+	zephyr_library_sources(devices/${MCUX_DEVICE}/system_MIMX8MN6_cm7.c)
```
Comment (galak) [inline mcux/CMakeLists.txt:65]: Can you move this down and make it similar to how the CONFIG_SOC_MIMXRT685S_CM33 is done.

**Review by galak — CHANGES_REQUESTED** [review]
Please also update `mcux/README`

Please also merge the commits so they look similar to:

8fd1f29b209b17695e50984641fe434fbcd73f7b

**Comment by waseemh40** [comment]
Updated README and formatted CMake file. Sorry, but I am not sure what do you mean by merging commits. Can you point to a stackoverflow link or the git commands used to merge commits? I only know about merging branches, but we have only 1 branch here. Do you want me to branch here with a new pull request with all the changes in a single commit message?

**Comment by galak** [comment]
> Updated README and formatted CMake file. Sorry, but I am not sure what do you mean by merging commits. Can you point to a stackoverflow link or the git commands used to merge commits? I only know about merging branches, but we have only 1 branch here. Do you want me to branch here with a new pull request with all the changes in a single commit message?

https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by galak** [comment]
> Updated README and formatted CMake file. Sorry, but I am not sure what do you mean by merging commits. Can you point to a stackoverflow link or the git commands used to merge commits? I only know about merging branches, but we have only 1 branch here. Do you want me to branch here with a new pull request with all the changes in a single commit message?

You can rebase your tree and as part of that squash the commits and rewrite the commit message similar to the commit I referenced.

**Comment by waseemh40** [comment]
OK. Do you want me to use single commit or 2 commits as proposed by you in the beginning?

**Comment by galak** [comment]
> OK. Do you want me to use single commit or 2 commits as proposed by you in the beginning?

2 would be great

File: mcux/CMakeLists.txt (hunk 1 above)
Comment (MaureenHelm) [reply mcux/CMakeLists.txt]: The drivers handling here needs to move to `mcux/drivers/imx8/CmakeLists.txt`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
There is a lot of extra stuff here that we don't usually import from the MCUXpresso SDK. Did you use the import script in `mcux/scripts/import_mcux_sdk.py`?

**Comment by waseemh40** [comment]
@galak Sorry for a delayed response/changes, was busy with another project. I have merged commits as per the requirements. I hope that it is NOT messed up, since I had to use force push at the end. Hope everything is in place now.

@MaureenHelm I have removed the drivers from mcux/CmakeLists.txt. Kindly see changes at the last commit.

**Comment by galak** [comment]
> @galak Sorry for a delayed response/changes, was busy with another project. I have merged commits as per the requirements. I hope that it is NOT messed up, since I had to use force push at the end. Hope everything is in place now.
> 

No worries, I think we are close to updating to HAL version 2.10.0 from NXP so need to see if we need to update your change to that HAL version

**Comment by galak** [comment]
Can you please update this to be based on SDK 2.10.0 (now that we merged that change)

**Comment by waseemh40** [comment]
OK. Will update to the newer SDK soon.

**Comment by dleach02** [comment]
@waseemh40, will you be returning to this PR?

**Pull request closed without merging**